source/daq/frequency_mask_trigger.cc

```diff
@@ -422,34 +435,65 @@ namespace psyllid
+                            if( t_mask_buffer.size() != t_array_size )
+                            {
+                                LWARN( plog, "Mask was not the right size; Resizing mask to " << t_array_size << " bins and filling it with zeros" );
+                                t_mask_buffer.resize( t_array_size, 0. );
+                            }
                             a_ctx.f_first_packet_after_start = false;
                         }
+
                         t_trigger_flag->set_flag( false );
+                        t_trigger_flag->set_high_threshold( false );
+                        t_trigger_flag->set_id( t_freq_data->get_pkt_in_session() );
 
                         for( unsigned i_bin = 0; i_bin < t_array_size; ++i_bin )
                         {
                             t_real = t_freq_data->get_array()[ i_bin ][ 0 ];
                             t_imag = t_freq_data->get_array()[ i_bin ][ 1 ];
-                /*#ifndef NDEBUG
-                            if( i_bin < 5 )
+                            t_power_amp = t_real*t_real + t_imag*t_imag;
+
+                            t_trigger_flag->set_id( t_freq_data->get_pkt_in_session() );
+
+                            if( high_threshold_factor == 1 )
                             {
-                                LWARN( plog, "Bin " << i_bin << " -- real = " << t_real << ";  imag = " << t_imag << ";  value = " << t_real*t_real + t_imag*t_imag <<
-                                        ";  mask = " << f_mask[ i_bin ] );
+                                if( t_power_amp >= t_mask_buffer[ i_bin ] )
+                                {
+                                    t_trigger_flag->set_flag( true );
+                                    t_trigger_flag->set_high_threshold( true );
+                                    LDEBUG( plog, "Data id <" << t_trigger_flag->get_id() << "> [bin " << i_bin << "] resulted in flag <" << t_trigger_flag->get_flag() << ">" << '\n' <<
+                                            "\tdata: " << t_power_amp << ";  mask1: " << t_mask_buffer[ i_bin ] );
+                                    break;
+                                }
                             }
-                #endif*/
-                            t_trigger_flag->set_id( t_freq_data->get_pkt_in_session() );
-                            if( t_real*t_real + t_imag*t_imag >= t_mask_buffer[ i_bin ] )
+                            else
                             {
-                                t_trigger_flag->set_flag( true );
-                                LTRACE( plog, "Data " << t_trigger_flag->get_id() << " [bin " << i_bin << "] resulted in flag <" << t_trigger_flag->get_flag() << ">" << '\n' <<
-                                       "\tdata: " << t_real*t_real + t_imag*t_imag << ";  mask: " << t_mask_buffer[ i_bin ] );
-                                break;
+                                if(  t_power_amp >= t_mask_buffer[ i_bin ] * high_threshold_factor )
```

Comment:
I think it would be better to pre-calculate a mask buffer array for the high threshold values.  It'll be faster than all of these extra multiplications.

Reply:
Ok. I ll add a second mask array. Another way would be to add a second fmt node. Is that possible?
The two fmts would both get the freq packets from the receiver and they would both send flags to the event builder. In the event builder there could be an and/or logic for flags from several fmt nodes.
However I guess that that could cause access conflicts to the frequency packets?

Reply:
Yes, this could definitely be done as a second mode.  You could have two `execute_mode_X()` functions that get called from the `execute()` function depending on which mode the node was configured in.  This would be the most efficient when running because it would avoid the extra check of the mode for every data object.